python-client/tests/communications/test_dto_parsing.py

```diff
@@ -0,0 +1,350 @@
+            elif func_argument.transformationFunction is not None:
+                assert isinstance(func_argument.transformationFunction, websocket.ArtifactRef)
+                assert func_argument.transformationFunction.project_key == TEST_PROJECT_KEY
+                assert func_argument.transformationFunction.id == TEST_FUNC_ARGUMENT_TRANSFORMATION_FUNCTION_ID
+            elif func_argument.float_arg is not None:
+                assert func_argument.float_arg == TEST_FUNC_ARGUMENT_FLOAT
+            elif func_argument.int_arg is not None:
+                assert func_argument.int_arg == TEST_FUNC_ARGUMENT_INT
+            elif func_argument.str_arg is not None:
+                assert func_argument.str_arg == TEST_FUNC_ARGUMENT_STR
+            elif func_argument.bool_arg is not None:
+                assert func_argument.bool_arg == TEST_FUNC_ARGUMENT_BOOL
+            elif func_argument.kwargs is not None:
+                # TODO: test for kwargs
+                pass
+            elif func_argument.args is not None:
+                assert len(func_argument.args) == 2
+                assert func_argument.args[0].is_none is False
+                assert func_argument.args[0].name == TEST_FUNC_ARGUMENTS_NAME1
+                assert func_argument.args[0].int_arg is not None
+                assert func_argument.args[0].int_arg == TEST_FUNC_ARGUMENT_INT
+                assert func_argument.args[1].is_none is False
+                assert func_argument.args[1].name == TEST_FUNC_ARGUMENTS_NAME2
+                assert func_argument.args[1].str_arg is not None
+                assert func_argument.args[1].str_arg == TEST_FUNC_ARGUMENT_STR
+            else:
+                # Should not have the other case
+                assert False
+
+
+@pytest.mark.parametrize(
+    "dataset_processing_function_dict",
+    [
+        {},
+        {
+            "slicingFunction": {"project_key": TEST_PROJECT_KEY, "id": TEST_FUNC_ARGUMENT_SLICING_FUNCTION_ID},
+        },
+        {
+            "transformationFunction": {
+                "project_key": TEST_PROJECT_KEY,
+                "id": TEST_FUNC_ARGUMENT_TRANSFORMATION_FUNCTION_ID,
+            },
+        },
+        {
+            "slicingFunction": {"project_key": TEST_PROJECT_KEY, "id": TEST_FUNC_ARGUMENT_SLICING_FUNCTION_ID},
+            "arguments": [],
+        },
+        {
+            "transformationFunction": {
+                "project_key": TEST_PROJECT_KEY,
+                "id": TEST_FUNC_ARGUMENT_TRANSFORMATION_FUNCTION_ID,
+            },
+            "arguments": [],
+        },
+    ],
+)
+def test_dataset_processing_function(dataset_processing_function_dict):
+    dataset_processing_function = websocket.DatasetProcessingFunction.parse_obj(dataset_processing_function_dict)
+    assert isinstance(dataset_processing_function, websocket.DatasetProcessingFunction)
+    if dataset_processing_function.slicingFunction is not None:
```

Comment:
I think it's cleaner to just have several small tests for earch type instead of a parameterized one with an "if-else"